### PR TITLE
MTL-1457 generate partially accurate ncn_metadata from shcd.json

### PR DESCRIPTION
**Pull request by jacobsalmela**
Signed-off-by: Jacob Salmela <jacob.salmela@hpe.com>

#### Summary and Scope

- Partially implements MTL-1457

##### Issue Type

- RFE Pull Request

Generates an `ncn_metadata.csv` file _with_ xnames, but _without_ MAC addresses (MACs are coming in a subsequent PR)

#### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I tested this on internal system (x) (if yes, please include results or a description of the test)
 
#### Idempotency
 
N/A
 
#### Risks and Mitigations
 
Low.

### CODE REVIEW

**Review by heemstra — APPROVED** [review]

File: cmd/shcd.go [hunk 1]
```diff
@@ -290,12 +314,109 @@ func (id Id) GenerateXname() (xn string) {
...
+		_, cabNum := utf8.DecodeRuneInString(cabString)
+
+		// Convert to an int
+		cabinet, err := strconv.Atoi(cabString[cabNum:])
+
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		// Strip the u
```
Comment (rustydb) [inline cmd/shcd.go:332]: For sanitizing?

Not nit picking, but the comment doesn't really say _why_.

File: cmd/shcd.go [hunk 2]
```diff
@@ -290,12 +314,109 @@ func (id Id) GenerateXname() (xn string) {
...
+				slot, err := strconv.Atoi(i)
+
+				if err != nil {
+					log.Fatalln(err)
+				}
+
+				bmcOrdinal = (slot % 4) + 1
+
+				// Dual node chassis - Apollo 6500 XL645D -- L == b1, R == b2
+			} else if id.Vendor == "hpe" {
```
Comment (rustydb) [inline cmd/shcd.go:353]: Do we have hpe anywhere as a global constant?

**Review by rustydb — APPROVED** [review]

File: cmd/shcd.go (hunk 1 above)
Comment (jacobsalmela) [reply cmd/shcd.go:332]: The schema requires an `int` here, which you can't convert with the "u" there.

File: cmd/shcd.go (hunk 2 above)
Comment (jacobsalmela) [reply cmd/shcd.go:353]: No, but good idea..

**Pull request merged**